src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementEventNormalizationService.cs

```diff
@@ -201,5 +189,70 @@ private Task<bool> ProcessErrorAsync(Exception ex, EventData data)
             var handled = _exceptionTelemetryProcessor.HandleException(ex, _log);
             return Task.FromResult(!handled);
         }
+
+        private IEnumerable<(string, IMeasurement)> CreateMeasurementIterator(JToken token, string partitionId, ConcurrentBag<Exception> exceptions, Func<Exception, EventData, Task<bool>> errorConsumer, EventData evt)
+        {
+            IEnumerator<Measurement> enumerable = null;
+            Action<Exception> storeNormalizedException = ex =>
+            {
+                var normalizationException = new NormalizationDataMappingException(ex);
+                if (errorConsumer(normalizationException, evt).ConfigureAwait(false).GetAwaiter().GetResult())
+                {
+                    exceptions.Add(normalizationException);
+                }
+            };
+
+            try
+            {
+                enumerable = _contentTemplate.GetMeasurements(token).GetEnumerator();
+            }
+            catch (Exception ex)
+            {
+                storeNormalizedException(ex);
+                yield break;
+            }
+
+            (string, IMeasurement) currentValue = (partitionId, null);
+            var shouldLoop = true;
+            var stopWatch = new Stopwatch();
+
+            while (shouldLoop)
+            {
+                try
+                {
+                    stopWatch.Start();
```

Comment:
We have some helper classes that simplify this.  We may need to port them from PaaS to OSS.